src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs

```diff
@@ -890,6 +900,31 @@ public void Match_Advanced(string pattern, string input, RegexOptions options, i
             }
         }
 
+        [Theory]
+        // Anchors
+        [InlineData(@"^.*", "abc", 0, true, true)]
+        [InlineData(@"^.*", "abc", 1, false, true)]
+        // Positive Lookbehinds
+        [InlineData(@"(?<=abc)def", "abcdef", 3, true, false)]
+        // Negative Lookbehinds
+        [InlineData(@"(?<!abc)def", "abcdef", 3, false, true)]
+        public void Match_StartatDiffersFromBeginning(string pattern, string input, int startat, bool expectedSuccessStartAt, bool expectedSuccessBeginning)
+        {
+            foreach (RegexOptions line in new[] { RegexOptions.None, RegexOptions.Singleline, RegexOptions.Multiline })
```

Comment:
Nit, it might have been nice to use the MemberData approach instead of these foreaches, so they are distinct tests and we get a more specific message if one fails.